Assets/Scripts/Networking/ConnectionHandler.cs

```diff
@@ -0,0 +1,134 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class ConnectionHandler : Photon.PunBehaviour {
+
+	[SerializeField] string gameVersion = "0";
+
+	public List<Selectable> connectionReqdObject = new List<Selectable>();
+	List<bool> croStartingStates = new List<bool>();
+
+	bool isInRoom = false, startMatch = false;
+
+	public TextTweening infoText;
+
+	public List<string> waitingTexts = new List<string>()
+	{
+		"Waiting for an idle god",
+		".Waiting for an idle god.",
+		"..Waiting for an idle god..",
+		"...Waiting for an idle god..."
+	};
+
+	public string connectionSuccessText;
+
+	#region Unity Callbacks
+
+	private void Awake()
+	{
+		PhotonNetwork.autoJoinLobby = false;
+
+		PhotonNetwork.automaticallySyncScene = true;
+	}
+
+	private void Start()
+	{
+		PhotonNetwork.ConnectUsingSettings(gameVersion);
+
+		foreach (var item in connectionReqdObject)
+		{	
+			croStartingStates.Add(item.interactable);
+			item.interactable = false;
+		}
+	}
+
+	private void Update()
+	{
+		if (isInRoom)
+		{
+			if (PhotonNetwork.room.PlayerCount == 2 && !startMatch)
+			{
+				infoText.SetText(connectionSuccessText);
+
+				startMatch = true;
+
+				//Start game
+			}
+		}
+	}
+
+	#endregion
+
+
+	#region Primary Methods
+
+	/// <summary>
+	/// Start the connection process. 
+	/// - If already connected, we attempt joining a random room
+	/// - if not yet connected, Connect this application instance to Photon Cloud Network
+	/// </summary>
+	public void Connect(string ip)
```

Comment:
If this object is responsible for managing network state, should it be a singleton?

Reply:
Doesn't need to be. It doesn't manage network state moment-to-moment, and has no need to remember the past; it queries state and changes at the moment it's needed.